src/backend/distributed/utils/enable_ssl.c

```diff
@@ -0,0 +1,443 @@
+/*-------------------------------------------------------------------------
+ *
+ * enable_ssl.c
+ *    UDF and Utilities for enabling ssl during citus setup
+ *
+ * Copyright (c) 2018, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/connection_management.h"
+#include "distributed/worker_protocol.h"
+#include "libpq/libpq.h"
+#include "miscadmin.h"
+#include "nodes/parsenodes.h"
+#include "postmaster/postmaster.h"
+#include "utils/guc.h"
+
+#ifdef USE_OPENSSL
```

Comment:
Might as well use `USE_SSL` here, rather than `USE_OPENSSL`… either that, or use `USE_OPENSSL` everywhere else.

At the moment these two `#define`s are identical, but I suppose in the future they may not be. The includes here seem pretty OpenSSL-specific, but the implementation code below does as well.

From PostgreSQL:

```c
/*
 * USE_SSL code should be compiled only when compiling with an SSL
 * implementation.  (Currently, only OpenSSL is supported, but we might add
 * more implementations in the future.)
 */
#ifdef USE_OPENSSL
#define USE_SSL
#endif
```

Reply:
I will standardise on USE_OPENSSL in that case.

Reply:
We chatted more on this in Slack and decided using `USE_SSL` most places and `USE_OPENSSL` to guard includes is fine, as that's what PostgreSQL seems to do (even though it doesn't actually make a lot of sense, as much of the code guarded by `USE_SSL` can be OpenSSL-specific, occasionally.